### DISCUSSION TITLE
Pre-/Post-hooks also executed for pure resource backups

### DISCUSSION THREAD

**Discussion by henkypunky** [vmware-tanzu/velero#3474] [Community support Q&A] [answered]
Hi guys,

just want to have a confirmation and maybe a reasoning:

Seems that Velero will always invoke hooks on Pods when annotations are present even for pure resource backups when no PVC snapshots are in play (--use-volume-snapshots=false). Can anybody please confirm and maybe explain why the hooks are still considered to be beneficial in this scenario? Thanks!

**Comment by ashish-amarnath** [comment, marked as answer]
@henkypunky Yes, per and post hooks are executed on pods whenever a pod is being backed up.
One explanation for why we might want to do that is, the hooks don't have to be restricted to performing operations to snapshot persistent volumes. They can be used, for example, to communicate with the application that it is being backed up, and when the backup process has completed.  So it would be useful to run pre and post hooks even when volume snapshots are disabled.

**Comment by ashish-amarnath** [comment]
Here is the code to confirm that the hooks are run during pod backup, regardless of whether or not volume snapshotting is enabled.
https://github.com/vmware-tanzu/velero/blob/main/pkg/backup/item_backupper.go#L123-L126
https://github.com/vmware-tanzu/velero/blob/main/pkg/backup/item_backupper.go#L205-L208
